model/riscv_platform.sail

```diff
@@ -243,6 +243,24 @@ function clint_store(addr, width, data) = {
     mtimecmp = vector_update_subrange(mtimecmp, 63, 32, sail_zero_extend(data, 32)); /* FIXME: Redundant zero_extend currently required by Lem backend */
     clint_dispatch();
     MemValue(true)
+  } else if addr == MTIME_BASE & 'n == 8 then {
+    if   get_config_print_platform()
+    then print_platform("clint<8>[" ^ BitStr(addr) ^ "] <- " ^ BitStr(data) ^ " (mtime)");
+    mtime = data;
```

Comment:
If this zero_extend is no longer required, can we also remove it from the other cases? Not necessarily as part of this PR, but would be good to clean up the FIXME.

Reply:
Yeah I don't think `vector_update_subrange` is needed either. Can go in a different MR though.